### PR TITLE
Add Oauth2 authentication

### PR DESCRIPTION
**Pull request by guibeira**
### what
Add Oauth2 authentication

### why
to improve security

### CODE REVIEW

File: sample_fast_api/apps/auth/authentication.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+from sample_fast_api.apps.users.models import User
+from sample_fast_api.apps.users.schemas import pwd_context
+from sample_fast_api.config import settings
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+
+def validate_token(token: str = Depends(oauth2_scheme)):
+    credentials_exception = HTTPException(
```
Comment (fvlima) [inline sample_fast_api/apps/auth/authentication.py]: Maybe this variable can stay outside of this method, to be created once at app loading

File: sample_fast_api/apps/auth/authentication.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        )
+
+        return {"access_token": access_token, "token_type": "bearer"}
+
+    def _create_access_token(self, data: dict, expires_delta: Optional[timedelta] = None):
+        to_encode = data.copy()
+        if expires_delta:
+            expire = datetime.utcnow() + expires_delta
+        else:
+            expire = datetime.utcnow() + timedelta(minutes=15)
```
Comment (fvlima) [inline sample_fast_api/apps/auth/authentication.py]: This `15` number can be a constant here at file scope or a envvar configuration

File: tests/apps/auth/test_authentication.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from collections import namedtuple
+
+from unittest.mock import patch
```
Comment (fvlima) [inline tests/apps/auth/test_authentication.py:2]: Did you install `pre-commit`? I think that this blank line between these imports it would be alerted

**Comment by fvlima** [comment]
Great @GuilhermeVBeira. Thanks for this PR. I think that some problem with tox/linter is happening because there is an error in the build https://travis-ci.org/github/fvlima/sample-fast-api/builds/732129233. Can you see this? It's interesting to run tox locally to see this problem

File: sample_fast_api/apps/auth/authentication.py (hunk 2 above)
Comment (guibeira) [reply sample_fast_api/apps/auth/authentication.py]: Sorry, I forgot to include from config

File: sample_fast_api/apps/auth/authentication.py (hunk 1 above)
Comment (guibeira) [reply sample_fast_api/apps/auth/authentication.py]: :+1:

**Comment by guibeira** [comment]
> Great @GuilhermeVBeira. Thanks for this PR. I think that some problem with tox/linter is happening because there is an error in the build https://travis-ci.org/github/fvlima/sample-fast-api/builds/732129233. Can you see this problem? It's interesting to run tox locally to see this problem

Of course, checking

**Comment by guibeira** [comment]
I opened another PR to solve the travis error
https://github.com/fvlima/sample-fast-api/pull/3

**Pull request merged**